### ISSUE TITLE
xiaomi redmi 3 note pro does not work

### ISSUE THREAD

**Issue by martinbonnin** [HearthSim/arcane_tracker#35]
There are a few reviews complaining about that.

**Comment by Thelasterick** [comment]
It also has issues with xiaomi redmi 3s. It does not upload the stats in track-o-bot.

**Comment by vbaryash** [comment]
I confirm that there is an issue with Xiaomi Redmi 3s. It may upload stats for 1 match, and than doesn't upload for other ones. Than when you close and reopen arcanetracker, it may upload stats for other 1 or 2 matches (but not every time) and than stop uploading again.

**Comment by Thelasterick** [comment]
^^ this is exactly what I'm experiencing :(

**Comment by martinbonnin** [comment]
Xiaomi devices have a strange way of handling the overlay permissions. Please read https://www.reddit.com/r/arcanetracker/comments/5nygi0/read_this_if_you_are_playing_on_a_xiaomi_device/

**Comment by Thelasterick** [comment]
Already tried the fix. Still not getting my stats uploaded. Any other suggestions?

**Comment by martinbonnin** [comment]
From some logs, it looks like the tracker believes the phone is not connected to internet. The code for this is pretty standard (see below). Does that happen both with wifi & mobile connectivity ?

        ConnectivityManager cm = (ConnectivityManager) ArcaneTrackerApplication.getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        return activeNetwork != null && activeNetwork.isConnectedOrConnecting();

**Comment by Thelasterick** [comment]
Yes, I'm unable to make it work for both wifi and data connection.

**Comment by Thelasterick** [comment]
Will this be fixed? Still encountering the issue.

**Comment by martinbonnin** [comment]
No update here unfortunately :-(. If you or someone you know is able to compile the app, it should be pretty easy to confirm if this is indeed a ConnectivityManager problem.
